arch/arm/src/imxrt/imxrt_flexio.h

```diff
@@ -0,0 +1,394 @@
+  FLEXIO_SHIFTER_STOP_BIT_HIGH    = 0x3u, /* Set shifter stop bit to logic high level. */
+};
+
+/* Define type of START bit configuration. */
+
+enum flexio_shifter_start_bit_e
+{
+  FLEXIO_SHIFTER_START_BIT_DISABLED_LOAD_DATA_ON_ENABLE = 0x0u, /* Disable shifter start bit, transmitter loads data on enable. */
+  FLEXIO_SHIFTER_START_BIT_DISABLED_LOAD_DATA_ON_SHIFT  = 0x1u, /* Disable shifter start bit, transmitter loads data on first shift. */
+  FLEXIO_SHIFTER_START_BIT_LOW                          = 0x2u, /* Set shifter start bit to logic low level. */
+  FLEXIO_SHIFTER_START_BIT_HIGH                         = 0x3u, /* Set shifter start bit to logic high level. */
+};
+
+/* Define FlexIO shifter buffer type */
+
+enum flexio_shifter_buffer_type_e
+{
+  FLEXIO_SHIFTER_BUFFER                     = 0x0u, /* Shifter Buffer N Register. */
+  FLEXIO_SHIFTER_BUFFER_BIT_SWAPPED         = 0x1u, /* Shifter Buffer N Bit Byte Swapped Register. */
+  FLEXIO_SHIFTER_BUFFER_BYTE_SWAPPED        = 0x2u, /* Shifter Buffer N Byte Swapped Register. */
+  FLEXIO_SHIFTER_BUFFER_BIT_BYTE_SWAPPED    = 0x3u, /* Shifter Buffer N Bit Swapped Register. */
+  FLEXIO_SHIFTER_BUFFER_NIBBLE_BYTE_SWAPPED = 0x4u, /* Shifter Buffer N Nibble Byte Swapped Register. */
+  FLEXIO_SHIFTER_BUFFER_HALF_WORD_SWAPPED   = 0x5u, /* Shifter Buffer N Half Word Swapped Register. */
+  FLEXIO_SHIFTER_BUFFER_NIBBLE_SWAPPED      = 0x6u, /* Shifter Buffer N Nibble Swapped Register. */
+};
+
+struct flexio_timer_config_s
+{
+  uint32_t trigger_select;
+  enum flexio_timer_trigger_polarity_e    trigger_polarity;
+  enum flexio_timer_trigger_source_e      trigger_source;
+  enum flexio_pin_config_e                pin_config;
+  uint32_t                                pin_select;
+  enum flexio_pin_polarity_e              pin_polarity;
+  enum flexio_timer_mode_e                timer_mode;
+  enum flexio_timer_output_e              timer_output;
+  enum flexio_timer_decrement_source_e    timer_decrement;
+  enum flexio_timer_reset_condition_e     timer_reset;
+  enum flexio_timer_disable_condition_e   timer_disable;
+  enum flexio_timer_enable_condition_e    timer_enable;
+  enum flexio_timer_stop_bit_condition_e  timer_stop;
+  enum flexio_timer_start_bit_condition_e timer_start;
+  uint32_t timer_compare;
+};
+
+struct flexio_shifter_config_s
+{
+  uint32_t timer_select;
+  enum flexio_shifter_timer_polarity_e    timer_polarity;
+  enum flexio_pin_config_e                pin_config;
+  uint32_t                                pin_select;
+  enum flexio_pin_polarity_e              pin_polarity;
+  enum flexio_shifter_mode_e              shifter_mode;
+  uint32_t                                parallel_width;
+  enum flexio_shifter_input_source_e      input_source;
+  enum flexio_shifter_stop_bit_e          shifter_stop;
+  enum flexio_shifter_start_bit_e         shifter_start;
+};
+
+/* The FlexIO vtable */
+
+struct flexio_dev_s;
+
+struct flexio_ops_s
+{
```

Comment:
same overly wrapped?

Reply:
OK, I see what you mean. I was using some editor plugin to automatically wrap the text and sometimes it is too agressive.